ape_hardhat/providers.py

```diff
@@ -347,10 +347,12 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
             receipt = self.get_receipt(
                 txn_hash.hex(), required_confirmations=txn.required_confirmations or 0
             )
-            receipt.raise_for_status()
+
+            if raise_on_fail:
+                receipt.raise_for_status()
 
         else:
-            receipt = super().send_transaction(txn)
+            receipt = super().send_transaction(txn, raise_on_fail=raise_on_fail)
```

Comment:
I know this is the ideal, but I am wondering in the interim if we can side-step having to pass in `raise_on_fail` to the super call by copying the logic directly here an putting a TODO to replace it with this approach after the next release.

I am trying to avoid having this be a breaking change.

Reply:
Marking as resolved since we're going to stage these changes for 0.6.0